### PR TITLE
Copy state in between hashing and init()

### LINKED ISSUE #7487: Make sure virtual map is mutable when init is called
Our current workflows for updating the application state at upgrade boundaries are a bit ad hoc. This is causing issues with undesired interactions between platform lifecycles and virtual map lifecycles. After a virtual map is hashed, it becomes immutable. But it needs to be mutable when init() is called.

### PR DESCRIPTION
**Pull request by cody-littley**
Closes #7487

### CODE REVIEW

**Review by lpetrovic05 — APPROVED** [review]

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/Browser.java [hunk 1]
```diff
@@ -737,6 +737,25 @@ private Collection<SwirldsPlatform> createLocalPlatforms(
         return Collections.unmodifiableList(platforms);
     }
 
+    /**
+     * Create a copy of the initial signed state. There are currently data structures that become immutable after
+     * construction, and we need to make a copy to force it to be mutable again.
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/Browser.java:742]: The way this description is worded makes it sound like the data structures in question become immutable *immediately* after construction, though I assume it just means "at some point after construction"?

Is there a way to make this more clear? A reader with the wrong interpretation might wonder how making a copy will help, if the structures become immutable after being constructed

**Review by alittley — APPROVED** [review]

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/Browser.java (hunk 1 above)
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/Browser.java:742]: Updated. Specifically, virtual maps become immutable after being hashed.

```
     * Create a copy of the initial signed state. There are currently data structures that become immutable after
     * being hashed, and we need to make a copy to force it to become mutable again.
```

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/Browser.java [hunk 2]
```diff
@@ -758,9 +776,6 @@ private static void updateLoadedStateAddressBook(
         new MerkleRouteIterator(state, state.getPlatformState().getAddressBook().getRoute())
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/Browser.java:776]: Is it still necessary to invalidate the hash down to the address book?

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/Browser.java [hunk 3]
```diff
@@ -758,9 +776,6 @@ private static void updateLoadedStateAddressBook(
...
                 .forEachRemaining(MerkleNode::invalidateHash);
 
-        // We should only have to rehash a few nodes, so simpler to use the synchronous algorithm.
-        MerkleCryptoFactory.getInstance().digestTreeSync(state);
-
         // If our hash changes as a result of the new address book then our old signatures may become invalid.
```
[the remark this answers is not in the document]
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/Browser.java:779]: Good point. We should do this after we rehash.

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/Browser.java (hunk 2 above)
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/Browser.java:776]: You are correct, not necessary. These nodes will be unhashed in this method.

**Review by alittley — APPROVED** [review]

**Review by alittley — APPROVED** [review]

**Review by edward-swirldslabs — APPROVED** [review]

**Comment by poulok** [comment]
Please remember to also fix in `develop`

**Pull request merged**